Samples/PushNotifications/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,195 @@
+                    args.retryCount);
+            }
+        });
+
+    auto result = co_await channelOperation;
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        // Caller's responsibility to keep the channel alive
+        co_return result.Channel();
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task = RequestChannelAsync();
+    if (task.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result = task.GetResults();
+    return result;
+}
+
+int main()
+{
+    // Initialize dynamic dependencies so we can consume the Windows App SDK APIs in the Windows App SDK framework package from this unpackaged app. 
+
+    // Major.Minor version, MinVersion=0 to find any framework package for this major.minor version
+    const UINT32 c_Version_MajorMinor{ 0x00040001 };
+    const PACKAGE_VERSION minVersion{};
+    RETURN_IF_FAILED(MddBootstrapInitialize(c_Version_MajorMinor, nullptr, minVersion));
+
+    PushNotificationActivationInfo info(PushNotificationRegistrationActivators::ProtocolActivator);
+    PushNotificationManager::RegisterActivator(info);
+
+   
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+    switch (kind)
+    {
+
+    case ExtendedActivationKind::Launch:
+    {
+
+        // request a channel
+        PushNotificationChannel channel = RequestChannel();
+
+        // register the chaneel
+        if (channel)
+        {
+            auto channelUri = channel.Uri();
+            DateTimeFormatter formater = DateTimeFormatter(L"on {month.abbreviated} {day.integer(1)}, {year.full} at {hour.integer(1)}:{minute.integer(2)}:{second.integer(2)}");
+
+            std::cout << "Channel Uri: " << winrt::to_string(channelUri.ToString()) << std::endl << std::endl;
+            std::wcout << L"Channel Uri will expire " << formater.Format(channel.ExpirationTime()).c_str() << std::endl;
+
+            // Register Push Event for Foreground
+            winrt::event_token token = channel.PushReceived([](const auto&, PushNotificationReceivedEventArgs const& args)
+                {
+                    auto payload = args.Payload();
+
+                    // Do stuff to process the raw payload
+                    std::string payloadString(payload.begin(), payload.end());
+                    std::cout << "Push notification content received from FOREGROUND: " << payloadString << std::endl << std::endl;
+                    args.Handled(true);
```

Comment:
I think this deserves a comment, saying why it is set to true (to prevent background activation).

Reply:
Done